### ISSUE TITLE
[uniongang] (updating) Login failed: Selector "a[href="/logout.php"]" didn't match

### ISSUE THREAD

**Issue by Kirich2** [Jackett/Jackett#15607]
### Have you checked our Troubleshooting page for your issue?

- [X] I have checked the Troubleshooting page

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Environment

```markdown
Info Starting Jackett v0.22.664
Info Environment version: 8.0.8 (C:\ProgramData\Jackett\)
Info OS version: Microsoft Windows NT 10.0.17763.0 (64bit OS)
Info Jackett variant: CoreWindows
Info Running in Docker: No
Info ThreadPool MaxThreads: 1023 workerThreads, 1000 completionPortThreads
Info App config/log directory: C:\ProgramData\Jackett
Info Using proxy: Http
Info Using FlareSolverr: http://srv-torrent:8191
Info Using HTTP Client: HttpWebClient2
```

### Description

When adding the uniongang private indexer and saving the settings, I get an error: "An error occurred while updating this indexer
Login failed: Selector "a[href="/logout.php"]" didn't match"

### Logged Error Messages

```text
2024-09-23 23:05:27.6048 Error Jackett.Common.ExceptionWithConfigData: Login failed: Selector "a[href="/logout.php"]" didn't match

[v0.22.664.0] Jackett.Common.ExceptionWithConfigData: Login failed: Selector "a[href="/logout.php"]" didn't match
   at Jackett.Common.Indexers.Definitions.CardigannIndexer.TestLogin() in ./Jackett.Common/Indexers/Definitions/CardigannIndexer.cs:line 908
   at Jackett.Common.Indexers.Definitions.CardigannIndexer.ApplyConfiguration(JToken configJson) in ./Jackett.Common/Indexers/Definitions/CardigannIndexer.cs:line 1067
   at Jackett.Server.Controllers.IndexerApiController.UpdateConfig(ConfigItem[] config) in ./Jackett.Server/Controllers/IndexerApiController.cs:line 97
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jackett.Server.Middleware.CustomExceptionHandler.Invoke(HttpContext httpContext) in ./Jackett.Server/Middleware/CustomExceptionHandler.cs:line 26
```

### Screenshots

![uniongang](https://github.com/user-attachments/assets/0c74d4d8-9bbb-4cdc-ab6e-99be8c9005bb)

**Comment by Kirich2** [comment]
In the described appeal 15584, the solution method is not described. The site is accessed completely normally. I also read the recommendations about the cookie mode, but I did not find any switch for this mode, and where to insert these cookies

**Comment by ilike2burnthing** [comment]
> Info Using proxy: Http

> The site is accessed completely normally.

Are you using the exact same HTTP proxy server (so, the same public IP address) when accessing the site in browser, as Jackett is using?

**Comment by garfield69** [comment]
The solution is to first login using your browser to ensure that the web site is not presenting a code challenge during the login.
Once you are logged in via the browser, you can then set up the indexer with your credentials, and the indexer should be able to login successfully.
The indexer is working for me.
The mention of cookies was a proposal in the event the indexer was no longer able to login, but this is not needed. To implement the cookie method we would need to rewrite the indexer.

**Comment by ilike2burnthing** [comment]
@garfield69 I can't say I missed the creepy psychic twin posting 😛

**Comment by Kirich2** [comment]
> > Информация об использовании прокси: Http
> 
> > Доступ к сайту осуществляется совершенно нормально.
> 
> Вы используете точно такой же HTTP-прокси-сервер (то есть тот же общедоступный IP-адрес) при доступе к сайту в браузере, что и Jackett?

Да прокси один и тот же, сайт работает в том же браузере нормально. Без прокси доступа нет, заблокировано.

**Comment by ilike2burnthing** [comment]
Please provide an enhanced log of the issue - https://github.com/Jackett/Jackett/wiki/Troubleshooting#how-to-provide-a-jackett-enhanced-log-for-a-ticket

**Comment by Kirich2** [comment]
[2024-09-23 233519.8945 Debug WebCli.txt](https://github.com/user-attachments/files/17104067/2024-09-23.233519.8945.Debug.WebCli.txt)

**Comment by ilike2burnthing** [comment]
Yea, definitely giving you that challenge.

So if you use the exact same proxy as Jackett for your browser, and open a new private/incognito window, with no browser addons/extensions enabled, and then attempt to sign into the tracker, it is successful without presenting you with a code challenge?

If so, try again with new proxy server, using the browser first (again though, it has to be the exact same server for both Jackett and the browser).

**Comment by Kirich2** [comment]
I use a translator, and maybe there are translation problems and I don't understand something, but: 1. I tried all the options with and without a proxy, incognito and with extensions disabled, with any options I can get to the site safely and log in, but jackett can't. 2. the proxy server is my personal one, there is no filtering, just forwarding

**Comment by ilike2burnthing** [comment]
K, I'll change the indexer to cookies login later, when I get a second. It will be available in the next release ~15.5hrs from now.

**Comment by Kirich2** [comment]
Hello, I'll be waiting. If I am required to provide anything, I am ready

**Comment by garfield69** [comment]
I disagree with changing the original indexer, since this is working for everyone else apart for the OP.
I will create a cookie version of indexer, as we have done for ygg and others.

**Comment by garfield69** [comment]
So with v0.22.671 uniongangcookie indexer is now available.
Give it a go and see if it helps you bypass your issue.

**Comment by Kirich2** [comment]
I've already lost hope. I tried with cookies, unfortunately it didn't work either. I've tried both the main .net site and the mirror.club, with and without proxy, in incognito mode and simply. I can get to the site everywhere, but I can't connect the indexer.  I tried another browser, but alas, the same error again. The only thing I noticed when I changed the browser was that the site was tracking presence. When I tried to log in to a new browser, it let me in and gave me an error that the user is already active, and the activity will reset after 10 minutes. This happened 10 minutes after the page was closed in the old browser. I tried to log out of all the pages and log in to Jacket after 10 minutes, but I failed here too.
